lib/src/arithmetic/fixed_point/fixed_point_value.dart

```diff
@@ -0,0 +1,46 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// fixed_point_value.dart
+//
+// 2024 September 24
+// Authors:
+//  Max Korbel <max.korbel@intel.com>
+//  Desmond A Kirkpatrick <desmond.a.kirkpatrick@intel.com
+//  Soner Yaldiz <soner.yaldiz@intel.com>
+
+import 'dart:math';
+import 'package:rohd/rohd.dart';
+import 'package:rohd_hcl/rohd_hcl.dart';
+
+/// A flexible representation of signed fixed-point values following Q notation
+/// (Qm.n format) as introduced by
+/// (Texas Instruments)[https://www.ti.com/lit/ug/spru565b/spru565b.pdf]
+class FixedPointValue implements Comparable<FixedPointValue> {
+  /// The full fixed point value bit storage in two's complement
+  late final LogicValue value;
+
+  /// The sign of the fixed point number
+  final LogicValue sign;
+
+  /// The integer part of the fixed point number
+  final LogicValue integer;
+
+  /// The fractional part of the fixed point number
+  final LogicValue fraction;
+
+  /// Constructor of a [FixedPointValue] from integer and fraction values
+  FixedPointValue(
+      {required this.sign, required this.integer, required this.fraction}) {
+    if (sign.isZero) {
```

Comment:
Is there a way to construct this class with one or two of these fields not provided:  e.g. no sign and no fraction would give you just a 32bit integer value  (347.)   or no sign and no integer would give you only positive fractions (.347).  I don't know if you want to generalize just yet, but figuring out the constructor would tell you if you even can handle it.
It looks like we support 0-length logic values.

Reply:
Great point, I forgot about integer only or fraction only scenarios and I can refactor accordingly. By definition, Q notation is for signed number. If the sign is optional, there should not be a reference to Q notation.

Reply:
For faster iterations, I am leaning towards these generalizations later.

Reply:
Either integer or fraction must be specified. sign is always required.